test/test_runner/test_parser.cpp

```diff
@@ -33,6 +34,17 @@ std::unique_ptr<TestGroup> TestParser::parseTestFile() {
     return std::move(testGroup);
 }
 
+void TestParser::genGroupName() {
+    std::size_t subStart =
+        TestHelper::appendKuzuRootPath(std::string(TestHelper::E2E_TEST_FILES_DIRECTORY)).length() +
+        1;
+    std::size_t subEnd = path.find_last_of('.') - 1;
+    std::string relPath = path.substr(subStart, subEnd - subStart + 1);
+    std::replace(relPath.begin(), relPath.end(), '-', '_');
```

Comment:
I don't think we should replace `-` with `_`. Users of the testing framework should follow the convention. themselves.

Reply:
I did this as a workaround as a bunch of files/directories had `-` in the codebase which results in a false positive during the test runs. I could also go and change those files and remove this line.

Reply:
What do u mean "false positive" here?

Reply:
Removed this line and fixed the naming of existing tests